WordPress/gutenberg
# Update the block theme folders to templates and parts

See #36548

This PR updates the block themes to use `templates` and `parts` instead of `block-templates` and `block-template-parts`. 
It also keeps backward compatibility, that said a theme can not have "templates" and "block-template-parts" or "block-templates" and "parts". I also updated all the documentation we have.

**Notes**

The issue also mentions the future of patterns and styles, I think these are not requirements for 5.9

**Testing instructions**

 - Make sure themes with "block-templates" and "block-template-parts" folders work properly.
 - Make sure themes with "templates" and "parts" folders work properly.
